azext_edge/edge/providers/check/mq.py

```diff
@@ -746,10 +1199,354 @@ def _evaluate_broker_diagnostics_service(
                     target_name=target_brokers,
                     namespace=namespace,
                     display=Padding(
-                        f"[cyan]{port.get('name')}[/cyan] "
-                        f"port [bright_blue]{port.get('port')}[/bright_blue] "
-                        f"protocol [cyan]{port.get('protocol')}[/cyan]",
+                        f"{colorize_string(port.get('name'))} "
+                        f"port {colorize_string(port.get('port'))} "
+                        f"protocol {colorize_string(port.get('protocol'))}",
                         (0, 0, 0, 16),
                     ),
                 )
             check_manager.add_display(target_name=target_brokers, namespace=namespace, display=NewLine())
+
+
+def _get_valid_references(kind: Union[Enum, str], namespace: Optional[str] = None) -> Dict[str, Any]:
+    result = {}
+    custom_objects = MQ_ACTIVE_API.get_resources(kind=kind, namespace=namespace)
+    if custom_objects:
+        objects: List[dict] = custom_objects.get("items", [])
+        for object in objects:
+            o: dict = object
+            metadata: dict = o.get("metadata", {})
+            name = metadata.get("name")
+            if name:
+                result[name] = True
+
+    return result
```

Comment:
This might be helpful if pulled into a base resource function that takes an API object, what do you think?